src/components/Chat/ChatDesktop.tsx

```diff
@@ -1,76 +1,158 @@
-import { LuSend } from "react-icons/lu";
 import { SlPlus } from "react-icons/sl";
-import { ChatBanner } from "./ChatBanner";
-import { ChatName } from "./ChatName";
-import { Navigation } from "./Navigation";
+import { ChatListItem } from "./ChatListItem";
+import { Navigation } from "./Navigation/Navigation";
+import DataContext, {
+  ChatDataTypes,
+  DataContextProps,
+} from "../../context/DataContext";
+import { useState, useContext, useEffect } from "react";
+import NewPersonalChatModal from "../Modals/NewPersonalChatModal";
+import ChatBox from "./ChatBox";
+import { SendersInfo } from "../../utils/SendersInfo";
+import SettingsModal from "../Modals/SettingsModal";
+
+export enum ModalsEnum {
+  SETTINGS = "settings",
+  NEW_PERSONAL_CHAT = "newPersonalChat",
+  NEW_GROUP_CHAT = "newGroupChat",
+  CHAT = "chat",
+  NOT_SHOW = "",
+}
 
 const ChatDesktop = () => {
+  const { setChats, chats, setSelectedChat, userData } = useContext(
+    DataContext
+  ) as DataContextProps;
+
+  const [selectedId, setSelectedId] = useState("");
+  const [personalChats, setPersonalChats] = useState<ChatDataTypes[]>([]);
+  const [modal, setModal] = useState(ModalsEnum.NOT_SHOW);
+
+  const modalView = () => {
+    switch (modal) {
+      case ModalsEnum.NEW_PERSONAL_CHAT:
+        return (
+          <NewPersonalChatModal
+            createNewPersonalChat={createNewPersonalChat}
+            setModal={setModal}
+          />
+        );
+      case ModalsEnum.SETTINGS:
+        return <SettingsModal setModal={setModal} />;
+      case ModalsEnum.NEW_GROUP_CHAT:
+        return (
+          <NewPersonalChatModal
+            createNewPersonalChat={createNewPersonalChat}
+            setModal={setModal}
+          />
+        );
+      case ModalsEnum.CHAT:
+        return (
+          <NewPersonalChatModal
+            createNewPersonalChat={createNewPersonalChat}
+            setModal={setModal}
+          />
+        );
+      default:
+        return null;
+    }
+  };
+
+  useEffect(() => {
+    // Update personalChats whenever chats changes
+    if (chats) {
+      const updatedPersonalChats = chats.filter(
+        ({ isGroupChat }) => !isGroupChat
+      );
+
+      setPersonalChats(updatedPersonalChats);
+    }
+  }, [chats]);
+
+  const createNewPersonalChat = (newChat: ChatDataTypes) => {
+    const doesChatExist = personalChats.some(
+      (item) => item._id === newChat._id
+    );
+
+    if (doesChatExist) {
+      const sender = SendersInfo(userData, newChat);
+      setSelectedChat(sender);
+      setSelectedId(newChat._id);
+      return;
+    } else {
```

Comment:
if you have return in previous if statement, then you do not need else key word.